File: backend/src/main/java/az/azcup/backend/exception/ConflictException.java
package az.azcup.backend.exception;

import org.springframework.http.HttpStatus;

// 409 Conflict — məs. "bu e-poçtla artıq hesab var" kimi hallarda atılır
// (bax: AuthService.register).
public class ConflictException extends ApiException {
    // Mesajı 409 CONFLICT statusu ilə birlikdə ApiException-a ötürür.
    public ConflictException(String message) {
        super(HttpStatus.CONFLICT, message);
    }
}

File: backend/src/main/kotlin/az/azcup/backend/exception/ConflictException.kt
package az.azcup.backend.exception

import org.springframework.http.HttpStatus

// 409 Conflict — məs. "bu e-poçtla artıq hesab var" kimi hallarda atılır
// (bax: AuthService.register).
class ConflictException(message: String) : ApiException(HttpStatus.CONFLICT, message)
